File: Offer/12/Solution.java
public class Solution {
    public double Power(double base, int exponent) {
        if(exponent == 0) return 1.0;
        if( exponent < 0 ){
            base = 1 / base;
            exponent = -exponent;
        }
        double res = Power(base, exponent / 2);
        if(exponent & 1){
            return res * res * b;
        }else{
            return res * res;
        }

    }
}

File: Offer/12/Solution.go
package Solution

func Power(b float64, n int) float64 {
	if n == 0 {
		return 1.0
	}
	if n < 0 {
		b = 1 / b
		n = -n
	}
	res := Power(b, n/2)
	if n%2 == 1 {
		return res * res * b
	} else {
		return res * res
	}
}
